docs/modules/ROOT/pages/how-tos/configuring/customizing-the-build.adoc

```diff
@@ -239,6 +336,90 @@ The task update PR will trigger the on-pull-request pipeline, testing the change
 
 By default, all pipelines push the images to a local repository that is set up as a part of installation. Ths registry address is registry-service.kind-registry:5001. It is not mandatory to use this local repo, so if you want to use your own Quay repo to control user permissions, you can do this by following link:https://github.com/konflux-ci/konflux-ci/blob/main/docs/quay.md#configuring-a-push-secret-for-the-build-pipeline[the instructions] for configuring a push secret for the build piepline.
 
+== Adding sidecar containers to the build pipeline
+
+You can add sidecar containers to the build pipeline to run additional tasks alongside the main container. For example, you can add a sidecar container to run a database to enable your unit tests to run against a real database.
+
+In the example, we add a sidecar container to task to run a PostgreSQL database. Such sidecar can be used to test database seeding, migrations, etc.
```

Comment:
Is the primary use case for this unit tests? If so, should we add this to the testing pages?

Reply:
> Is the primary use case for this unit tests?

Correct.